src/vs/workbench/test/browser/workbenchTestServices.ts

```diff
@@ -944,6 +974,36 @@ export class RemoteFileSystemProvider implements IFileSystemProvider {
 	private toFileResource(resource: URI): URI { return resource.with({ scheme: Schemas.file, authority: '' }); }
 }
 
+export class TestInMemoryFileSystemProvider extends InMemoryFileSystemProvider implements IFileSystemProviderWithFileReadStreamCapability {
+	readonly capabilities: FileSystemProviderCapabilities =
+		FileSystemProviderCapabilities.FileReadWrite
+		| FileSystemProviderCapabilities.PathCaseSensitive
+		| FileSystemProviderCapabilities.FileReadStream;
+
+	readFileStream(resource: URI): ReadableStreamEvents<Uint8Array> {
+		const stream = newWriteableStream<Uint8Array>(data => VSBuffer.concat(data.map(data => VSBuffer.wrap(data))).buffer);
+
+		this.readFile(resource)
+			.then(data => {
+				for (let i = 0; i < data.length; i++) {
+					const CHUNK_SIZE = 128;
```

Comment:
I selected this chunk size randomly without any rationale. Do you think there is a better choice?

Reply:
Maybe use the same we use in our disk based file system provider?

https://github.com/microsoft/vscode/blob/129cc5a66898829340df56f6abb86e0171f5b30e/src/vs/platform/files/node/diskFileSystemProvider.ts#L47